### ISSUE TITLE
[Feature]: add llama guard support 

### ISSUE THREAD

**Issue by krrishdholakia** [BerriAI/litellm#2030]
### The Feature

support moderation checks with llama guard from meta on sagemaker 

### Motivation, pitch

user request

**Comment by ghost** [comment]
LLM checks will significantly slow down responses. We'll probably need to expose a new `async_moderation_hook(..)` 

Can scope it to just `/chat/completions` for now. 

This hook can be run in parallel to the llm api call, but both calls (actual llm api call + moderation check) must be completed before returning a response to the user.

**Comment by ghost** [comment]
Further improvements: 
- enable API callbacks for `async_moderation_hook(..)`

**Comment by ghost** [comment]
How a user might set this up: 

```yaml
litellm_settings:
   callbacks: ["llamaguard_moderations"]
   llamaguard_model_name: "sagemaker/jumpstart-dft-meta-textgeneration-llama-guard-7b"
```

**Comment by ghost** [comment]
Merged into main. Should be live soon in v`1.25.0`.
